src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -0,0 +1,340 @@
+            SERVICE_ERROR_NORMAL = 0x00000001,
+            SERVICE_ERROR_SEVERE = 0x00000002,
+            SERVICE_ERROR_CRITICAL = 0x00000003
+        }
+
+        /// <summary>
+        /// Describes the configuration information to be changed.
+        /// </summary>
+        public enum ServiceInfoLevel
+        {
+            SERVICE_CONFIG_DESCRIPTION = 1,
+            SERVICE_CONFIG_FAILURE_ACTIONS = 2,
+            SERVICE_CONFIG_DELAYED_AUTO_START_INFO = 3,
+            SERVICE_CONFIG_FAILURE_ACTIONS_FLAG = 4,
+            SERVICE_CONFIG_SERVICE_SID_INFO = 5,
+            SERVICE_CONFIG_REQUIRED_PRIVILEGES_INFO = 6,
+            SERVICE_CONFIG_PRESHUTDOWN_INFO = 7,
+            SERVICE_CONFIG_TRIGGER_INFO = 8,
+            SERVICE_CONFIG_PREFERRED_NODE = 9,
+            SERVICE_CONFIG_LAUNCH_PROTECTED = 12
+        }
+
+        /// <summary>
+        /// Changes the optional configuration parameters of a service.
+        /// </summary>
+        /// <param name="hService">
+        /// A handle to the service.
+        /// This handle is returned by the OpenService or CreateService function and
+        /// must have the <see cref="ServiceAccess.SERVICE_CHANGE_CONFIG"/> access right.
+        /// </param>
+        /// <param name="dwInfoLevel">
+        /// The configuration information to be changed.
+        /// This parameter can be one value from <see cref="ServiceStartType"/>.
+        /// </param>
+        /// <param name="lpInfo">
+        /// A pointer to the new value to be set for the configuration information. 
+        /// The format of this data depends on the value of the dwInfoLevel parameter. 
+        /// If this value is NULL, the information remains unchanged.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, the return value is nonzero.
+        /// If the function fails, the return value is zero
+        /// </returns>
+        [DllImport(nameof(AdvApi32), SetLastError = true, CharSet = CharSet.Unicode)]
+        public static extern bool ChangeServiceConfig2(SafeServiceHandler hService, ServiceInfoLevel dwInfoLevel, IntPtr lpInfo);
+
+        /// <summary>
+        /// Closes a handle to a service control manager or service object.
+        /// </summary>
+        /// <param name="hSCObject">
+        /// A handle to the service control manager object or the service object to close.
+        /// Handles to service control manager objects are returned by the <see cref="OpenSCManager"/> function,
+        /// and handles to service objects are returned by either the <see cref="OpenService"/> or <see cref="CreateService"/> function.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, the return value is nonzero.
+        /// If the function fails, the return value is zero.
+        /// </returns>
+        [DllImport(nameof(AdvApi32), SetLastError = true)]
+        private static extern bool CloseServiceHandle(IntPtr hSCObject);
```

Comment:
Now that you've made this private (thanks!), please move it down below all your public methods to keep StyleCop warnings happy.